### PR TITLE
Update Lighhouse architecture.

### PR DESCRIPTION
**Pull request by aswinsuryan**
Signed-off-by: Aswin Surayanarayanan <asuryana@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,33 +6,33 @@ The below digram shows the basic Lighthouse architecture.
...
 
-The Lighthouse controller will be running along with the [Admiral](https://github.com/submariner-io/admiral) control plane. It uses the API’s exposed by the Admiral to connect to each cluster and listen for Service CRUD. Once it is notified that a new Service is created and needs to be distributed, it creates the Lighthouse CRD and the same is distributed to all the clusters that are part of the control plane. The flow for update and delete will be similar.
+The lighthouse agent will get updated, whenever a new lighthouse CRD is created or deleted in the broker.
 
 ### WorkFlow
-The typical workflow is as follows.
+The workflow is as follows,
+
+- Lighthouse agent connects to the kube-api-server of the broker.
+- It creates Lighthouse CRD for every service in the local cluster.
```
Comment (mangelajo) [inline README.md:18]:
```suggestion
- It creates MultiClusterService CR for every service in the local cluster.
```

File: README.md [hunk 2]
```diff
@@ -6,33 +6,33 @@ The below digram shows the basic Lighthouse architecture.
...
+- It syncs the Lighthouse CRD to and from the broker.
```
Comment (mangelajo) [inline README.md:19]:
```suggestion
- It syncs the  MultiClusterService CR to and from the broker.
```

File: README.md [hunk 3]
```diff
@@ -6,33 +6,33 @@ The below digram shows the basic Lighthouse architecture.
 
 ![Lighthouse Architecture](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/architecture.png)
 
-## Lighthouse Controller
-This is the central discovery controller that gathers the information from the clusters, decides what information is to be shared and distributes the information as newly defined CRDs (Kubernetes custom resources).
+## Lighthouse Agent
+The Lighthouse Agent runs in every cluster and it has access to the Kubernetes api-server running in the broker. It creates a lighthouse crd for each service and sync the CRD with broker. It also retrieves the information about services running in another cluster from the broker and creates a lighthouse CRD locally.
```
Comment (tpantelis) [inline README.md:10]:
```suggestion
The Lighthouse Agent runs in every cluster and accesses the Kubernetes API server running in the broker cluster to exchange service metadata information with other clusters. Local service information is exported to the broker and service information from other clusters is imported. 
```

File: README.md [hunk 4]
```diff
@@ -6,33 +6,33 @@ The below digram shows the basic Lighthouse architecture.
...
 
-The Lighthouse controller will be running along with the [Admiral](https://github.com/submariner-io/admiral) control plane. It uses the API’s exposed by the Admiral to connect to each cluster and listen for Service CRUD. Once it is notified that a new Service is created and needs to be distributed, it creates the Lighthouse CRD and the same is distributed to all the clusters that are part of the control plane. The flow for update and delete will be similar.
+The lighthouse agent will get updated, whenever a new lighthouse CRD is created or deleted in the broker.
```
Comment (tpantelis) [inline README.md:12]: Remove this line.

File: README.md [hunk 5]
```diff
@@ -6,33 +6,33 @@ The below digram shows the basic Lighthouse architecture.
...
 
 ### WorkFlow
-The typical workflow is as follows.
+The workflow is as follows,
+
+- Lighthouse agent connects to the kube-api-server of the broker.
```
Comment (tpantelis) [inline README.md:17]:
```suggestion
- Lighthouse agent connects to the broker's K8s API server.
```

File: README.md (hunk 1 above)
Comment (tpantelis) [reply README.md:18]:
```suggestion
- For every service in the local cluster, it creates a corresponding MultiClusterService resource and exports it to the broker to be consumed by other clusters.
```

File: README.md (hunk 2 above)
Comment (tpantelis) [reply README.md:19]:
```suggestion
- For every MultiClusterService resource in the broker exported from another concert, it creates a copy of it in the local cluster.
```

File: README.md [hunk 6]
```diff
@@ -6,33 +6,33 @@ The below digram shows the basic Lighthouse architecture.
...
+- It creates Lighthouse CRD for every service in the local cluster.
+- It syncs the Lighthouse CRD to and from the broker.
 
-- Lighthouse controller registers with Admiral for cluster join and unjoin events.
-- Once it is notified about a join event, it retrieves the credentials from Admiral and registers for a call back on Service creation with a shared informer.
-- When a new Service is created in a cluster, the Lighthouse controller will be notified about it and it will create the Lighthouse CRD with the Service info.
-- The controller with the help of Admiral distributes the CRD to all the clusters that have joined.
 
 ![Lighthouse Controller WorkFlow](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/controllerWorkFlow.png)
```
Comment (tpantelis) [inline README.md:22]:
```suggestion
![Lighthouse Agent WorkFlow](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/controllerWorkFlow.png)
```

File: README.md [hunk 7]
```diff
@@ -6,33 +6,33 @@ The below digram shows the basic Lighthouse architecture.
...
 
 ## Lighthouse Plugin
-Lighthouse can be installed as an external plugin for CoreDNS, and will work along with the default Kubernetes plugin. It uses the Lighthouse CRD that is distributed by the controller for DNS resolution. The below diagram indicates a high-level architecture.
+Lighthouse can be installed as an external plugin for CoreDNS, and will work along with the default Kubernetes plugin. It uses the Lighthouse CRD that is created by the lighthouse agent for DNS resolution. The below diagram indicates a high-level architecture.
```
Comment (tpantelis) [inline README.md:25]:
```suggestion
Lighthouse can be installed as an external plugin for CoreDNS, and will work along with the default Kubernetes plugin. It uses the MultiClusterService resource that is created by the lighthouse agent for DNS resolution. The below diagram indicates a high-level architecture.
```

**Comment by aswinsuryan** [comment]
Have removed the architecture section and provided the link to https://github.com/submariner-io/submariner-website/pull/92

**Comment by aswinsuryan** [comment]
> Would it be possible to just remove, not update, the binary image file? IMHO we should try to avoid binary files in this code-focused repo (compared to docs-focused, like the website) as each version will forever live in the git commit history, eventually making it large. Maybe we can just depend on the diagram on the website?

Removed the files. It was not required anymore.

**Pull request merged**